### PR TITLE
refactor: extract buildLevels function and improve node grouping logic

### PR DESCRIPTION
**Pull request by toyamarinyon**
## Summary
Refactors node grouping and execution level building logic for better modularity and maintainability.

### Code Organization
- **Extract `buildLevels` function**: Move inline implementation from `createAct` to dedicated `build-levels.ts` module
- **Improve `groupNodes` return structure**: Change from `NodeGroup[]` to object with separate `operationNodeGroups` and `triggerNodeGroups`
- **Add `findNodeGroupByNodeId` helper**: Simplify finding node groups by node ID

### API Simplification  
- **Use `nodeId` instead of `connectionIds`**: Simplify `createAct` and GitHub event handler APIs
- **Streamline `use-node-groups` hook**: Remove duplicated logic by directly using `groupNodes`

### Testing & Documentation
- **Add comprehensive test suite**: 474 lines of tests for `buildLevels` with edge cases and real-world scenarios
- **Add complex test fixture**: `twoTriggerFixture` representing a real GitHub PR QA workflow with 1124 lines
- **Update existing tests**: Adapt to new `groupNodes` return structure

### Benefits
- Better separation of concerns between level building and node grouping
- Improved reusability of `buildLevels` function
- More robust handling of trigger vs operation node groups  
- Enhanced test coverage for complex workflow scenarios
- Cleaner API surface with `nodeId` parameter instead of derived `connectionIds`

## Testing
- All existing functionality preserved
- Added extensive test coverage for extracted functions
- Updated integration tests to use new API patterns

## Summary by CodeRabbit

* **New Features**
  * Added a comprehensive workflow fixture supporting dual GitHub triggers and parallel QA/commenting flows for pull request automation.

* **Bug Fixes**
  * Improved handling of node grouping and execution levels, ensuring correct grouping of operation and trigger nodes.

* **Refactor**
  * Simplified node group and execution level utilities for clearer separation between operation and trigger nodes.
  * Streamlined related React hooks to leverage new grouping logic.
  * Updated event handling to use direct node IDs instead of grouped connection IDs.

* **Tests**
  * Introduced extensive tests for node grouping and execution level utilities.
  * Updated existing tests to match new group structure and function behaviors.

### CODE REVIEW

**Comment by toyamarinyon** [comment]
i'll fix build failure

File: packages/giselle/src/engine/utils/workspace/group-nodes.ts [hunk 1]
```diff
@@ -71,5 +86,68 @@ export function groupNodes(
...
+	for (const group of nodeGroups) {
+		const nodes = group.nodeIds
+			.map((nodeId) => workspace.nodes.find((node) => node.id === nodeId))
+			.filter((node) => node !== undefined);
+		const connections = group.connectionIds
+			.map((connectionId) =>
+				workspace.connections.find(
+					(connection) => connection.id === connectionId,
+				),
+			)
```
[the remark this answers is not in the document]
Comment (toyamarinyon) [reply packages/giselle/src/engine/utils/workspace/group-nodes.ts:105]: Thank you for the performance optimization suggestion. I've carefully considered the proposed Map-based approach and decided to keep the current
  implementation for the following reasons:

  1. Performance is not a concern at this scale
  - The .find() operations run on already-grouped data, not the entire workspace
  - After Union-Find grouping, we typically have 1-3 groups with just a few nodes each
  - Our test fixtures show realistic workspace sizes of ~10 nodes and ~8 connections

  2. Follows our "Less is more" philosophy
  - The current code is direct and obvious: workspace.nodes.find(node => node.id === nodeId)
  - Adding Maps would introduce unnecessary abstraction without solving a real problem
  - We prefer the simplest data structures that work for the given scale

  3. No measurable impact
  - The theoretical O(n) vs O(1) difference is negligible for our use case
  - The Map creation overhead (O(n+m)) would likely offset any gains for small datasets
  - We haven't identified this as a bottleneck through profiling

  If we encounter performance issues with large workspaces (hundreds of nodes) in the future, we'll profile first and optimize where it actually matters. For
  now, the current implementation strikes the right balance between simplicity and performance.

  The code is working as intended and is appropriately optimized for our current needs.
